File: ClassProject/Ex078ViewBinding-20220324T070849Z-001/Ex078ViewBinding/app/src/main/java/com/mrhi2022/ex078viewbinding/MyAdapter.java
package com.mrhi2022.ex078viewbinding;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.mrhi2022.ex078viewbinding.databinding.RecyclerItemBinding;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.VH> {

    Context context;
    ArrayList<Item> items;

    public MyAdapter(Context context, ArrayList<Item> items) {
        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView= LayoutInflater.from(context).inflate(R.layout.recycler_item, parent, false);
        return new VH(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {

//        Item item= items.get(position);
//        holder.binding.tv.setText(item.title);
//        holder.binding.iv.setImageResource(item.imgId);

        holder.bindItem(items.get(position));

    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class VH extends RecyclerView.ViewHolder{

        //recycler_item.xml 과 연결을 담당하는 바인딩클래스가 자동으로 존재함
        //이 바인딩 클래스는 xml 문서의 이름을 기반으로 클래스이름이 만들어짐.
        RecyclerItemBinding binding;

        public VH(@NonNull View itemView) {
            super(itemView);
            binding= RecyclerItemBinding.bind(itemView);
        }

        void bindItem(Item item){
            binding.iv.setImageResource(item.imgId);
            binding.tv.setText(item.title);
        }
    }
}

File: ClassProject/Ex101NaverSearchAPI/app/src/main/java/com/wookie_soft/ex101naversearchapi/adapters/MyAdapter.kt
package com.wookie_soft.ex101naversearchapi.adapters

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.text.HtmlCompat
import androidx.exifinterface.R
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.wookie_soft.ex101naversearchapi.databinding.RecylerItemBinding
import com.wookie_soft.ex101naversearchapi.model.SearchItem

class MyAdapter(val context: Context, val items:MutableList<SearchItem>) : RecyclerView.Adapter<MyAdapter.VH>() {

    inner class VH(itemView: View) :RecyclerView.ViewHolder(itemView){
        val binding:RecylerItemBinding = RecylerItemBinding.bind(itemView)

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        val itemView:View = LayoutInflater.from(context).inflate(com.wookie_soft.ex101naversearchapi.R.layout.recyler_item, parent, false)// falde :  뷰홀더가붙여줄거라
        return VH(itemView)
    }

    override fun onBindViewHolder(holder: VH, position: Int) {

        // 그리고 가끔 웹을 크롤링 하면 태그문이 포함되어 있음.
        // 태그문을 없에자.
        var title = HtmlCompat.fromHtml(items[position].title, HtmlCompat.FROM_HTML_MODE_COMPACT)



        // holder안에 모든 내용아ㅣ 있음
        //holder.binding.tv.text = items.get(position).title
        holder.binding.tv.text = title
        holder.binding.lowPrice.text = items[position].lprice
        holder.binding.tvBrand.text = items[position].brand
        Glide.with(context).load(items[position].image).into(holder.binding.iv)

        holder.binding.root.setOnClickListener {
            val intent:Intent = Intent(Intent.ACTION_VIEW , Uri.parse(items[position].link)) // ACTION_VIEW : 화면만 가지면 무조건 띄우는 녀석
            context.startActivity(intent)
        }





    }

//    override fun getItemCount(): Int {
//        return items.size // 이 갯수대로 만들어짐 !!
//    }

    override fun getItemCount(): Int = items.size
}